src/Protocol/DFRN.php

```diff
@@ -3089,4 +3089,73 @@ private static function constructVerb(array $item)
 		}
 		return ACTIVITY_POST;
 	}
+
+	private static function tgroupCheck($uid, $item)
+	{
+		$mention = false;
+
+		// check that the message originated elsewhere and is a top-level post
+
+		if ($item['wall'] || $item['origin'] || ($item['uri'] != $item['parent-uri'])) {
+			return false;
+		}
+
+		$u = q("SELECT * FROM `user` WHERE `uid` = %d LIMIT 1",
+			intval($uid)
+		);
+		if (!DBM::is_result($u)) {
+			return false;
+		}
+
+		$community_page = ($u[0]['page-flags'] == PAGE_COMMUNITY);
+		$prvgroup = ($u[0]['page-flags'] == PAGE_PRVGROUP);
+
+		$link = normalise_link(System::baseUrl() . '/profile/' . $u[0]['nickname']);
+
+		/*
+		 * Diaspora uses their own hardwired link URL in @-tags
+		 * instead of the one we supply with webfinger
+		 */
+		$dlink = normalise_link(System::baseUrl() . '/u/' . $u[0]['nickname']);
+
+		$cnt = preg_match_all('/[\@\!]\[url\=(.*?)\](.*?)\[\/url\]/ism', $item['body'], $matches, PREG_SET_ORDER);
+		if ($cnt) {
+			foreach ($matches as $mtch) {
+				if (link_compare($link, $mtch[1]) || link_compare($dlink, $mtch[1])) {
+					$mention = true;
+					logger('mention found: ' . $mtch[2]);
+				}
+			}
+		}
+
+		if (!$mention) {
+			return false;
+		}
+
+		return $community_page || $prvgroup;
+	}
+
+	/**
+	 * This function returns true if $update has an edited timestamp newer
+	 * than $existing, i.e. $update contains new data which should override
+	 * what's already there.  If there is no timestamp yet, the update is
+	 * assumed to be newer.  If the update has no timestamp, the existing
+	 * item is assumed to be up-to-date.  If the timestamps are equal it
+	 * assumes the update has been seen before and should be ignored.
+	 *
+	 */
+	private static function editedTimestampIsNewer($existing, $update)
```

Comment:
Please rename to `isEditedTimestampNewer`.

Reply:
Done